pkg/controller/backup_sync_controller.go

```diff
@@ -107,4 +120,32 @@ func (c *backupSyncController) run() {
 			}
 		}
 	}
+
+	err = deleteUnused(c, cloudBackups)
+	if err != nil {
+		c.logger.WithError(errors.WithStack(err)).Error("Error deleting unused backup from Kubernetes")
+	}
+}
+
+// deleteUnused deletes a backup from Kubernetes if it does not exist in object storage
+func deleteUnused(c *backupSyncController, cloudBackups map[string]*api.Backup) error {
+	// Backups from Kubernetes
+	backups, err := c.backupLister.Backups(c.namespace).List(labels.Everything())
+	if err != nil {
+		return err
```

Comment:
use `errors.WithStack(err)` - anywhere we're returning an error from non-Ark code, we wrap it using `errors.WithStack()` or `errors.Wrap()` which gives us a stack trace when it gets logged.